src/main/java/uk/gov/ons/ctp/response/action/service/impl/decorator/BusinessActionRequestDecoratorContextFactoryImpl.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.ons.ctp.response.action.service.impl.decorator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.response.action.domain.model.Action;
+import uk.gov.ons.ctp.response.action.service.PartySvcClientService;
+import uk.gov.ons.ctp.response.party.representation.Association;
+import uk.gov.ons.ctp.response.party.representation.PartyDTO;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO.SampleUnitType;
+
+@Slf4j
+@Component
+@Qualifier("business")
+public class BusinessActionRequestDecoratorContextFactoryImpl
+    extends ActionRequestDecoratorContextFactoryImpl {
+
+  @Autowired private PartySvcClientService partySvcClientService;
+
+  @Override
+  public ActionRequestDecoratorContext getActionRequestDecoratorContext(Action action) {
+    ActionRequestDecoratorContext context = super.getActionRequestDecoratorContext(action);
+
+    final PartyDTO parentParty;
+    List<PartyDTO> childParties;
+
+    if (context.getSampleUnitType().isParent()) {
+      parentParty =
```

Comment:
Maybe two named methods would make this code more self-explanatory

Reply:
Done